internal/characters/xiangling/guoba.go

```diff
@@ -88,21 +89,28 @@ func (p *panda) HandleAttack(atk *combat.AttackEvent) float64 {
 }
 
 func (p *panda) Attack(atk *combat.AttackEvent, evt glog.Event) (float64, bool) {
-	//don't take damage, trigger swirl reaction only on sucrose E
-	if p.Core.Player.Chars()[atk.Info.ActorIndex].Base.Key != keys.Sucrose {
-		return 0, false
-	}
 	if atk.Info.AttackTag != combat.AttackTagElementalArt {
 		return 0, false
 	}
-	//check pyro window
-	if p.Durability[reactable.ModifierPyro] < reactable.ZeroDur {
+	// check pyro window
+	if p.Durability[reactable.ModifierPyro] <= reactable.ZeroDur {
 		return 0, false
 	}
 
-	p.Core.Log.NewEvent("guoba hit by sucrose E", glog.LogCharacterEvent, p.c.Index)
+	// don't take damage, trigger swirl reaction only on sucrose E or faruzan E
+	switch p.Core.Player.Chars()[atk.Info.ActorIndex].Base.Key {
+	case keys.Sucrose:
+		p.Core.Log.NewEvent("guoba hit by sucrose E", glog.LogCharacterEvent, p.c.Index)
+	case keys.Faruzan:
+		if atk.Info.Abil != faruzan.SkillName {
```

Comment:
`SkillName` and `faruzan e` in the logs is ambigious considering that her pressurized collapse can be triggered via multiple ways. Please consider renaming this variable and the log below to indicate that it's the Pressurized Collapse.